internal/app/app_test.go

```diff
@@ -0,0 +1,88 @@
+package app_test
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"github.com/ubuntu/ubuntu-proxy-manager/internal/app"
+	"github.com/ubuntu/ubuntu-proxy-manager/internal/testutils"
+)
+
+func TestApp(t *testing.T) {
+	tests := map[string]struct {
+		applyArgs    []string
+		noSystemBus  bool
+		noMethodCall bool
+
+		wantNewErr      bool
+		wantDbusCallErr bool
+		wantRetCode     int
+	}{
+		"Empty apply arguments":   {},
+		"Correct apply arguments": {applyArgs: []string{"http://proxy:3128", "", "", "", "", ""}},
+		"No method called on app": {noMethodCall: true},
+
+		"Error when system bus is not available": {noSystemBus: true, wantNewErr: true},
+	}
+
+	for name, tc := range tests {
+		tc := tc
+		name := name
+		t.Run(name, func(t *testing.T) {
+			if !tc.noSystemBus {
+				defer testutils.StartLocalSystemBus()()
+			}
+
+			if len(tc.applyArgs) == 0 {
+				tc.applyArgs = []string{"", "", "", "", "", ""}
```

Comment:
I’m not a big fan: there is no type control because you are then casting that to a slice of interfaces, and what if you add one argument one day? It seems to be a recipe for issues, hard to debug tests and so on…